movingaverage/movingaverage.go

```diff
@@ -19,350 +19,134 @@ limitations under the License.
+	defaultBufferLength = 10
+)
 
-func newmovingAverage(length int) *average {
-	initCounter := 0
-	return &average{
-		movingAverageBuf: make([]interface{}, length),
-		counter:          initCounter,
-	}
+type average struct {
+	// buffer with past values
+	buffer []float64
+	// index of current element
+	counter int
+	// indicator if our buffer was already filled
+	full bool
 }
 
-//movingAverageProcessor is a struct which has a map that acts like a buffer for storage of values for different namespace
-// key is a namespace (type: string)
-//value is a pointer average struct which stores values of the namespace key
-type movingAverageProcessor struct {
+// MAProcessor struct hold map with average object for each unique namespace passed to plugin
+type MAProcessor struct {
 	movingAverageMap map[string]*average
-	movingBufLength  int
+	bufferLength     int
+	initialized      bool
 }
 
-//Each Namespace would have its own buffer-length and counter . Counter is used for the purpose of
-//replacing the oldest (when buffer is full) with the new value using mod operation
-type average struct {
-	movingAverageBuf []interface{}
-	counter          int
-}
-
-//Gets the current counter for the particular namespace
-func (p *movingAverageProcessor) getCounter(namespace string) (int, error) {
-	if _, ok := p.movingAverageMap[namespace]; ok {
-		return p.movingAverageMap[namespace].counter, nil
-	} else {
-		return -1, errors.New("Namespace is not present in the map")
+// New returns initialized plugin instance
+func New() *MAProcessor {
+	return &MAProcessor{
+		movingAverageMap: make(map[string]*average),
+		bufferLength:     defaultBufferLength,
+		initialized:      false,
 	}
 }
 
-//Sets the counter for the particular namespace
-func (p *movingAverageProcessor) setCounter(namespace string, counter int) error {
-	if _, ok := p.movingAverageMap[namespace]; ok {
-		p.movingAverageMap[namespace].counter = counter
-		return nil
-	} else {
-		return errors.New("Namespace is not present in the map")
+func newAverage(length int) *average {
+	avg := average{
+		buffer:  make([]float64, length),
+		counter: 0,
+		full:    false,
 	}
+	return &avg
 }
 
-func (p *movingAverageProcessor) getBufferLength() int {
-	return p.movingBufLength
-}
-
-func (p *movingAverageProcessor) setBufferLength(length int) error {
-	p.movingBufLength = length
-	return nil
-}
-
-//Adds data in the buffer for a particular namespace
-func (p *movingAverageProcessor) addBufferData(index int, data interface{}, namespace string) error {
-	if _, ok := p.movingAverageMap[namespace]; ok {
-		if index >= len(p.movingAverageMap[namespace].movingAverageBuf) {
-			return errors.New("Incorrect value of index, trying to access non-existing element of buffer")
-		}
-		p.movingAverageMap[namespace].movingAverageBuf[index] = data
-		return nil
-	} else {
-		return errors.New("Namespace is not present in the map")
+func castToFloat(data interface{}) (float64, error) {
```

Comment:
Ok, please add comment with this information to "protect" against unwanted changes.